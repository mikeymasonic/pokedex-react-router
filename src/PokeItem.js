import React, { Component } from 'react';
import './App.css';

export default class PokeItem extends Component {
    render() {
        return (
            <li>
            <div className="info-container">
                <h2>{this.props.pokemon.pokemon}</h2>
                <p className="poke-hp"><b>HP:</b> {this.props.pokemon.hp} / <b>XP:</b> {this.props.pokemon.base_experience}</p>
            </div>

            <div className="image-container">
                <img
                    alt={this.props.pokemon.url_image}
                    src={this.props.pokemon.url_image} />
            </div>
            
            <p className="attributes-class">
                <span id="type" style={{ backgroundColor: this.props.pokemon.color_1}}><b>Type: {this.props.pokemon.type_1}</b> </span><br></br>
                <b>Attack:</b> {this.props.pokemon.attack} / 
                <b> Defense:</b> {this.props.pokemon.defense}<br></br>
                <b> Special Attack:</b> {this.props.pokemon.special_attack} /
                <b> Special Defense:</b> {this.props.pokemon.special_defense}<br></br>
                <b> Speed:</b> {this.props.pokemon.speed}<br></br>
                <b> Height:</b> {this.props.pokemon.height}" / 
                <b> Weight:</b> {this.props.pokemon.weight} oz
            </p>
        </li>

            
        );
    }
}