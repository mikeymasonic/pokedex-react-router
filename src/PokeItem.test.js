import React from 'react';
import PokeItem from './PokeItem';
import renderer from "react-test-renderer";
test('renders PokeItem.js correctly', () => {
  const tree = renderer
  .create(<PokeItem pokeDex={[]} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});