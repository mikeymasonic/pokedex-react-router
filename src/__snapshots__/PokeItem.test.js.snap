// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PokeItem.js correctly 1`] = `
<li>
  <div
    className="info-container"
  >
    <h2 />
    <p
      className="poke-hp"
    >
      <b>
        HP:
      </b>
       
       / 
      <b>
        XP:
      </b>
       
    </p>
  </div>
  <div
    className="image-container"
  >
    <img />
  </div>
  <p
    className="attributes-class"
  >
    <span
      id="type"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <b>
        Type: 
      </b>
       
    </span>
    <br />
    <b>
      Attack:
    </b>
     
     /
    <b>
       Defense:
    </b>
     
    <br />
    <b>
       Special Attack:
    </b>
     
     /
    <b>
       Special Defense:
    </b>
     
    <br />
    <b>
       Speed:
    </b>
     
    <br />
    <b>
       Height:
    </b>
     
    " /
    <b>
       Weight:
    </b>
     
     oz
  </p>
</li>
`;
